rosidl_generator_cpp/resource/msg__struct.hpp.em

```diff
@@ -149,31 +149,46 @@ def generate_zero_string(membset, fill_args):
 @[if not member_list]@
     (void)_init;
 @[end if]@
-@[for membset in member_list]@
-@[ if membset.members[0].default_value is not None]@
+@{
+members_if_all_defaults = [m for m in member_list if m.members[0].default_value]
+members_else_if_zero = [m for m in member_list if m.members[0].zero_value]
+members_if_all_zero = [
+    m for m in member_list if  (m.members[0].zero_value or m.members[0].zero_need_array_override)
```

Comment:
Nitpick: two consecutive spaces. Alse wrap before the `if` (and if still necessary after the `and` instad of before).